src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -2,47 +2,231 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using Internal.LowLevelLinq;
+
 namespace System.Threading
 {
     internal static partial class ClrThreadPool
     {
         private const int CpuUtilizationHigh = 95;
         private const int CpuUtilizationLow = 80;
         private static int s_cpuUtilization = 85; // TODO: Add calculation for CPU utilization
-        
-        private static int s_minThreads; // TODO: Initialize
-        private static int s_maxThreads; // TODO: Initialize
 
-        // TODO: SetMinThreads and SetMaxThreads need to be synchronized with a lock
-        // TODO: Compare with CoreCLR implementation and ensure this has the same guarantees.
-        public static bool SetMinThreads(int threads)
+        private static short s_forcedMinWorkerThreads = 0; // TODO: Config
+        private static short s_forcedMaxWorkerThreads = 0; // TODO: Config
+
+        private const short MaxPossibleThreadCount = short.MaxValue;
+        private static short s_minThreads = (short)ThreadPoolGlobals.processorCount;
+        private static short s_maxThreads = short.MaxValue;
+        private static readonly LowLevelLock s_maxMinThreadLock = new LowLevelLock();
+
+        [Runtime.InteropServices.StructLayout(Runtime.InteropServices.LayoutKind.Explicit)]
+        private struct CacheLineAligned
+        {
+            [Runtime.InteropServices.FieldOffset(0)]
+            public ThreadCounts counts;
+            [Runtime.InteropServices.FieldOffset(16)]
+            public int lastDequeueTime;
+            [Runtime.InteropServices.FieldOffset(24)]
+            public int priorCompletionCount;
+        }
+
+        private static CacheLineAligned s_aligned = new CacheLineAligned();
+        private static int s_priorCompletedWorkRequestsTime;
+        private static int s_nextCompletedWorkRequestsTime;
```

Comment:
The two fields above can be in the same cache line as priorCompletionCount (they are read and written together)